src/overlays/actors/ovl_Bg_Jya_Megami/z_bg_jya_megami.c

```diff
@@ -21,31 +37,325 @@ const ActorInit Bg_Jya_Megami_InitVars = {
+        }
+        func_8002F974(&this->dyna.actor, NA_SE_EV_FACE_CRUMBLE_SLOW - SFX_FLAG);
+    } else if (this->lightTimer > 0) {
+        this->lightTimer--;
+    }
+    if (this->lightTimer > 40) {
+        Flags_SetSwitch(globalContext, this->dyna.actor.params & 0x3F);
+        BgJyaMegami_SetupExplosion(this);
+        Audio_PlaySoundAtPosition(globalContext, &this->dyna.actor.posRot.pos, 100, NA_SE_EV_FACE_EXPLOSION);
+        func_800800F8(globalContext, 0xD70, -0x63, &this->dyna.actor, 0);
+    } else {
+        if (this->lightTimer < 8) {
+            this->crumbleIndex = 0;
+        } else if (this->lightTimer < 16) {
+            this->crumbleIndex = 1;
+        } else if (this->lightTimer < 24) {
+            this->crumbleIndex = 2;
+        } else if (this->lightTimer < 32) {
+            this->crumbleIndex = 3;
+        } else {
+            this->crumbleIndex = 4;
+        }
+        CollisionCheck_SetAC(globalContext, &globalContext->colChkCtx, &this->collider.base);
+    }
+}
+
+void BgJyaMegami_SetupExplosion(BgJyaMegami* this) {
+    u32 i;
+
+    this->actionFunc = BgJyaMegami_Explosion;
+    for (i = 0; i < BGJYAMEGAMI_NUMPIECES; ++i) {
+        Math_Vec3f_Copy(&this->pieces[i].pos, &this->dyna.actor.posRot.pos);
+        this->pieces[i].vel.x = sPiecesInit[i].velX;
+    }
+    this->explosionTimer = 0;
+}
+
+void BgJyaMegami_Explosion(BgJyaMegami* this, GlobalContext* globalContext) {
+    BgJyaMegami_Piece* temp;
+    u32 i;
+    Vec3f sp8C;
+    BgJyaMegami_PieceInit* temp2;
+    u32 pad;
+
+    this->explosionTimer++;
+    if (this->explosionTimer == 30) {
+        Audio_PlaySoundAtPosition(globalContext, &this->dyna.actor.posRot.pos, 100, NA_SE_EV_FACE_BREAKDOWN);
+    }
+
+    for (i = 0; i < BGJYAMEGAMI_NUMPIECES; ++i) {
+        temp = &this->pieces[i];
+        temp2 = &sPiecesInit[i];
+        if (this->explosionTimer > temp2->delay) {
+            temp->vel.y -= 0.6f;
+            if (temp->vel.y < -18.0f) {
+                temp->vel.y = -18.0f;
+            }
+            temp->vel.x *= 0.995f;
+            temp->pos.x += temp->vel.x;
+            temp->pos.y += temp->vel.y;
+            temp->rotVelX += temp2->rotVelX;
+            temp->rotVelY += temp2->rotVelY;
+            if (Math_Rand_ZeroOne() < 0.067f) {
+                Math_Vec3f_Sum(&temp->pos, &temp2->unk_00, &sp8C);
+                sp8C.z += 10.0f;
+                func_8089A1DC(globalContext, &sp8C, &temp->vel, 3, 2);
+            }
+        } else if (this->explosionTimer == temp2->delay) {
+            Math_Vec3f_Sum(&temp->pos, &temp2->unk_00, &sp8C);
+            sp8C.z += 10.0f;
+            func_8089A1DC(globalContext, &sp8C, &temp->vel, 4, 2);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/func_8089A1DC.s")
+    if ((this->explosionTimer % 4 == 0) && (this->explosionTimer > 30) && (this->explosionTimer < 80) &&
+        (this->explosionTimer > 40)) {
```

Comment:
the redundancy of these checks makes me think that the constants 30 and 40 were originally defines, and because they are also magic constants elsewhere. 30 is the number of frames between the explosion beginning and the correct chime playing. 40 is the number of frames of light the face needs before it explodes (its "health"). Is it worth making these defines now?

Reply:
id say probably not for now, but curious what others think